es6/lib/createRouteHandler.js

```diff
@@ -0,0 +1,137 @@
+import forEach from 'lodash/collection/forEach';
+import isEmpty from 'lodash/lang/isEmpty';
+import mapKeys from 'lodash/object/mapKeys';
+import omit from 'lodash/object/omit';
+import rearg from 'lodash/function/rearg';
+import snakeCase from 'lodash/string/snakeCase';
+
+function apiController(options, knexClient) {
+  return {
+
+    create(request, reply) {
+      var resource;
+
+      if (options.tableHeaderFormat === 'snakeCase') {
+        var snakeCaseKeys = rearg(snakeCase, [1, 0]);
+        resource = mapKeys(request.payload, snakeCaseKeys);
+      } else {
+        resource = request.payload;
+      }
+
+      function createApiResponse(response) {
+        let apiResponse = { };
+        apiResponse[camelCase(options.index)] = first(response);
+        return apiResponse;
+      }
+
+      return knexClient
+        .table(options.tableName)
+        .insert(resource)
+        .then(createApiResponse)
+        .then(reply)
+        .catch(reply);
+    },
+
+    destroy(request, reply) {
+      function createDeleteResponse(result) {
+        if (result === 1 || result === 0) {
+          reply().code(204);
+          return;
+        }
+        reply().code(500);
+      }
+
+      return knexClient
+        .table(options.tableName)
+        .where(options.index, request.params.id)
+        .del()
+        .then(createDeleteResponse)
+        .catch(reply);
+    },
+
+    healthcheck(request, reply) {
+      reply({ status: 'ok', version: options.version });
+    },
+
+    list(request, reply) {
+      function filterQuery() {
+        function constructWhere(value, key) {
+          if (options.tableHeaderFormat === 'snakeCase') {
+            this.where(snakeCase(key), value);
+          } else {
+            this.where(key, value);
+          }
+        }
+
+        forEach(omit(request.query, ['cursor', 'limit']), constructWhere, this);
+      }
+
+      function prepareResponse(result) {
+        var camelCaseKeys = rearg(camelCase, [1, 0]);
+        return mapKeys(result, camelCaseKeys);
+      }
+
+      function createApiResponse(result) {
+        console.log(result.length);
+
+        var apiResponse = {
+          limit: null,
+          cursor: null,
+          records: result
+        };
+
+        if (isEmpty(result)) {
+          reply(apiResponse);
+          return;
+        }
+
+        apiResponse.limit = request.query.limit;
+
+        if (result.length === request.query.limit) {
+          apiResponse.cursor = request.query.cursor + 1;
+        }
+
+        reply(apiResponse);
+      }
+
+      return knexClient
+        .table(options.tableName)
+        .where(filterQuery)
+        .limit(request.query.limit)
+        .offset((request.query.cursor - 1) * request.query.limit)
+        .map(prepareResponse)
+        .then(createApiResponse)
+        .catch(reply);
+    },
+
+    show(request, reply) {
+      function createApiResponse(result) {
+        var camelCaseKeys;
```

Comment:
You can't hoist the declaration without pulling the definition with it, but you can do `const foo = 'bar';` at the top of a block along with the `var` statements.